pages/login.js

```diff
@@ -55,35 +66,65 @@ export default function Login() {
     const onSubmitLoginForm = async () => {
         setIsTryingToSignIn(true);
         setErrorMessage(null);
-        await auth.signin(name, password, homeserver).catch(/** @param {MatrixError} error */(error) => {
-            if (error.errcode === 'M_LIMIT_EXCEEDED') setErrorMessage(t('{{error}}, please wait {{time}} seconds.', { error: error.data.error, time: Math.round(error.data.retry_after_ms / 1000) }));
-            else setErrorMessage(error.data?.error);
-        });
+
+        if (auth.user) {
+            await auth.validateAuthProvidersAccessTokens(
+                auth.getAuthenticationProvider('matrix').getMatrixClient().getUserIdLocalpart(),
+                password,
+            );
+        } else {
+            await auth.signin(name, password, homeserver).catch(/** @param {MatrixError} error */(error) => {
+                if (error.errcode === 'M_LIMIT_EXCEEDED') {
+                    setErrorMessage(t('{{error}}, please wait {{time}} seconds.', {
+                        error: error.data.error,
+                        time: Math.round(error.data.retry_after_ms / 1000),
+                    }));
+                } else setErrorMessage(error.data?.error);
+            });
+        }
```

Comment:
All this looks like another leftover from the old forward-aroo concept. Can you otherwise explain it to me?

Reply:
ah yeah, thought this was for validating a session if a user is already logged in. but that doesn't happen here...